packages/lesswrong/components/seasonal/meetupMonth/MeetupMonthBanner.tsx

```diff
@@ -0,0 +1,541 @@
+    },
+    '&:hover': {
+      opacity: 1,
+    },
+    cursor: 'pointer',
+  },
+  contentContainer: {
+    width: 'calc(100% - 400px)',
+    [theme.breakpoints.up(smallBreakpoint)]: {
+      width: 'calc(100% - 300px)',
+    },
+    paddingTop: 120,
+    paddingBottom: 80,
+    position: 'absolute',
+    zIndex: 1,
+    right: 0,
+    bottom: 0,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'flex-end',
+    alignItems: 'center',
+  },
+  createEventButton: {
+    ...theme.typography.commentStyle,
+    color: theme.palette.primary.dark,
+    paddingTop: 10,
+    fontSize: 15,
+    display: 'inline-block',
+    cursor: 'pointer',
+    border: 'none',
+  },
+  createEventButtonIcon: {
+    // display: 'none',
+    fontSize: 16,
+    fontWeight: 900,
+    marginRight: 8,
+    marginLeft: 8,
+  },
+  acxMode: {
+    '& svg': {
+      fill: theme.palette.meetupMonth.acx + ' !important',
+    },
+  },
+  ifanyoneMode: {
+    '& svg': {
+      fill: theme.palette.meetupMonth.ifanyone + ' !important',
+    },
+  },
+  petrovMode: {
+    '& svg': {
+      fill: theme.palette.meetupMonth.petrov + ' !important',
+    },
+  },
+}));
+
+
+export const MeetupMonthQuery = gql(`
+  query meetupMonthQuery {
+    HomepageCommunityEvents(limit: 500) {
+      events {
+        _id
+        lat
+        lng
+        types
+      }
+    }
+  }
+`);
+
+
+export default function MeetupMonthBannerInner() {
+  const classes = useStyles(styles);
+  const [isLoading, setIsLoading] = useState(true);
+  const [everClickedMap, setEverClickedMap] = useState(false);
+  const carouselSections = useMemo(() => getCarouselSections(classes), [classes]);
+  
+  const defaultViewport = useMemo(() => ({
+    latitude: 20, 
+    longitude: -70,
+    zoom: 1.1
+  }), [])
+
+  const [viewport, setViewport] = useState(defaultViewport)
+
+  useEffect(() => {
+    const initializeMap = async () => {
+      setViewport({
+        latitude: defaultViewport.latitude,
+        longitude: defaultViewport.longitude,
+        zoom: defaultViewport.zoom
+      });
+      setIsLoading(false);
+    };
+
+    void initializeMap();
+  }, [defaultViewport]);
+
+  const { data } = useQuery(MeetupMonthQuery)
+
+  const events = useMemo(() => data?.HomepageCommunityEvents.events ?? [], [data?.HomepageCommunityEvents.events]);
+
+  const [isTransitioning, setIsTransitioning] = useState(false)
+  const [isSettingUp, setIsSettingUp] = useState(false)
+  const [currentCarouselIndex, setCurrentCarouselIndex] = useState(0)
+  const [nextCarouselIndex, setNextCarouselIndex] = useState<number | null>(null)
+
+  
+  const acxCarouselIndex = 1 
+  const acxActive = (nextCarouselIndex ?? currentCarouselIndex) === acxCarouselIndex
+  const ifanyoneCarouselIndex = 2 
+  const ifanyoneActive = (nextCarouselIndex ?? currentCarouselIndex) === ifanyoneCarouselIndex
+  const petrovCarouselIndex = 3 
+  const petrovActive = (nextCarouselIndex ?? currentCarouselIndex) === petrovCarouselIndex
```

Comment:
Ok clearly `nextCarouselIndex ?? currentCarouselIndex` needs to be assigned to its own variable, right

Reply:
(also these are clearly mututally exclusive and shouldn't be booleans but should be `type ActiveCarouselType = 'acx' | 'ifanyone' | 'petrov';`)